retriever/engines/mysql.py

```diff
@@ -39,6 +42,37 @@ class engine(Engine):
                       "{db}.{table}"),
                      ]
 
+    def check_credentials(self):
+        self.get_input()
+        try:
+            if self.opts["user"] != "root":
```

Comment:
This check is not needed `if self.opts["user"] != "root":`,
The logic can be summarized and below is a description of a better design for both postgres and mysql


I don't think we need a default dictionary,
``sql_lookup = {"user": "root", "password": ""}``
If we do need one, we should include a default port number and host.

We have 3 variables at this point in the code that we could use optimally.
We know the user, path to the config and required_opts file because self.get_input() does most of the initialization.

All we have to do is check if config file is there, if yes check if our user's credentials are in there.
Get the credentials and overwrite the current opts values. (This is like a greedy approach but I do think it's enough)


Given a command ``retriever install (engine_name either [postgres\mysql]) [options] iris`` with options as the examples below:

```
	retriever install postgres  -u henry 
	retriever install postgres iris 
	retriever install postgres –-user myusername –-password ***** –-host localhost –-port 5432
	retriever install mysql –-user myusername –-password ***** –-host localhost –-port 8888 –-database_name testdbase breed-bird-survey
```

we can say

If we do have a config file, like this(if file does exist)


```
hostname:port:database:root:password1
:::henry:password2
hostname:3333:database:travis:password3
hostname:4322:database:postgres:password4
hostname:port:database:apveyor:password5
hostname:port:database:apveyor:password5
hostname:port:database:apveyor:password5
```


```
	Read the lines, split and get the user
	If the User is same as what we have in the self.opts["user"](if our user is Travis)
	     # update  self.opts. 
	      For example in postgres, 
	       I can say if self.opts["user"] = line.slip(":")[the user user index](line =hostname:3333:database:v:password3)
	      do a greedy updatge and say since user credentials just update self.opts with new port number, and password and              host(from line get the user's port number, password, host)
	```

Why I am calling it greedy.

A user can have many configurations like the user `apveyor` in our example config file above.(I just would take the first read of a line having a user)
Also some users may decide to have just password and username, :::henry:password2